src/proto/h1/decode.rs

```diff
@@ -26,7 +28,8 @@ pub(crate) struct Decoder {
     kind: Kind,
 }
 
-#[derive(Debug, Clone, Copy, PartialEq)]
+// FIXME: can we keep Copy trait?
```

Comment:
is it fine to drop `Copy` here? this type is internal and nothing seemed to break.

Reply:
It's fine, I think. If the compiler doesn't complain then we're not really using `Copy` anywhere internally.